extensions/sql-migration/src/api/utils.ts

```diff
@@ -1169,3 +1171,83 @@ export async function clearDropDown(dropDown: DropDownComponent): Promise<void>
 	await dropDown.updateProperty('value', undefined);
 	await dropDown.updateProperty('values', []);
 }
+
+export async function getRecommendedConfiguration(targetType: MigrationTargetType, model: MigrationStateModel): Promise<string[]> {
+
+	if (!hasRecommendations(model)) {
+		return [];
+	}
+	let recommendation;
+	switch (targetType) {
+		case MigrationTargetType.SQLVM:
+			// elastic model currently doesn't support SQL VM, so show the baseline model results regardless of user preference
+			recommendation = model._skuRecommendationResults.recommendations?.sqlVmRecommendationResults[0];
+
+			// result returned but no SKU recommended
+			if (!recommendation?.targetSku) {
+				return [constants.SKU_RECOMMENDATION_NO_RECOMMENDATION];
+			}
+			else {
+				const vmConfiguration = constants.VM_CONFIGURATION(
+					recommendation.targetSku.virtualMachineSize!.sizeName,
+					recommendation.targetSku.virtualMachineSize!.vCPUsAvailable);
+
+				const dataDisk = constants.STORAGE_CONFIGURATION(
+					recommendation.targetSku.dataDiskSizes![0].size,
+					recommendation.targetSku.dataDiskSizes!.length);
+				const storageDisk = constants.STORAGE_CONFIGURATION(
+					recommendation.targetSku.logDiskSizes![0].size,
+					recommendation.targetSku.logDiskSizes!.length);
+				const tempDb = recommendation.targetSku.tempDbDiskSizes!.length > 0
+					? constants.STORAGE_CONFIGURATION(
+						recommendation.targetSku.logDiskSizes![0].size,
+						recommendation.targetSku.logDiskSizes!.length)
+					: constants.LOCAL_SSD;
+				const vmConfigurationPreview =
+					constants.VM_CONFIGURATION_PREVIEW(dataDisk, storageDisk, tempDb);
+
+				return [vmConfiguration, vmConfigurationPreview];
+			}
+		case MigrationTargetType.SQLDB:
+			const recommendations = model._skuEnableElastic
+				? model._skuRecommendationResults.recommendations!.elasticSqlDbRecommendationResults
+				: model._skuRecommendationResults.recommendations!.sqlDbRecommendationResults;
+			const successfulRecommendationsCount = recommendations.filter(r => r.targetSku !== null).length;
```

Comment:
can recommendations.filter().length return null and throw?  
I recommend a pattern like this:
const value = recommendations.filter()?.length ?? 0;

Reply:
done